File: JavaScriptPractice/Leetcode392.js
var isSubsequence = function(s, t) {
    let slow=0;
    let fast=0;
    if (s.length==0) {
        return true
    }
    if (t.length==0) {
        return false
    }
    while (fast<t.length) {
        if (s[slow]==t[fast]) {
            fast++
            slow++
        }
        else 
        {
            fast++
        }
        if (slow>=s.length) {
            return true
        }
        
    }
    return false
};

File: CSharpPractice/LeetcodePractice/Leetcode392.cs
public class Solution392 {
    public bool IsSubsequence(string s, string t) {
        int fast = 0;
        int slow=0;
        if (s.Length==0)
        {
            return true;
        }
        if (t.Length==0)
        {
            return false;
        }
        while (fast<t.Length)
        {
            if (s[slow]==t[fast])
            {
                slow++;
                fast++;
            }
            else
            {
                fast++;
            }
            if (slow>=s.Length)
            {
                return true;
            }
        }
        return false;
        
    }
}